### PR TITLE
:memo: Describe repo organization and processes

### PR DESCRIPTION
**Pull request by joerunde**
This PR adds a new TGIS specific readme with some docs about what this repo is, the diffs it contains from vLLM, and instructions on how to manage our contributions.

### CODE REVIEW

File: TGIS_README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+See a sketch of the commit graph
+![vllm commit strategy](docs/source/assets/tgis-vllm-repo.png)
+
+## Contributing changes for vLLM
+
+To contribute vLLM-specific changes, _don't_ base them on the `main` branch in this repo.
+Either contribute to your own fork of vLLM, or create a branch in this repo at the latest `vllm-project/vllm:main` commit.
+
+Open all PRs into [vLLM](https://github.com/vllm-project/vllm)
```
Comment (njhill) [inline TGIS_README.md:20]: We may want to dilute the language here a bit, something along the lines of "assuming it's an improvement to vLLM that would have general applicability/utility to the community" (though this should be the case most of the time).

File: TGIS_README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Once you have opened a PR, follow the steps for [Reconstructing the ephemeral branch](#Ephemeral-branch-building-procedure)
+
+## Contributing changes specific to the TGIS adapter or IBM delivery
+
+Contributing changes to the TGIS adapter or IBM delivery process is business as usual, 
+make changes on a branch or fork directly from this repo's `main` branch and PR it back in.
```
Comment (njhill) [inline TGIS_README.md:27]: May want to mention here that from a practical pov it probably makes most sense to branch from `ephemeral` for dev/testing purposes, but when ready we would merge instead into `main`.

File: TGIS_README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+# rebase the branch onto ibm main
+git rebase origin/main
+# You may see _many_ warnings about skipping previously applied commits- this is expected
+
+# Push to origin/main
+git push origin HEAD:main
+```
+
+## Ephemeral branch building procedure
```
Comment (njhill) [inline TGIS_README.md:62]: Still not sure about the `ephemeral` name for this. Perhaps something like `release` would be better?

File: TGIS_README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+To rebuild the ephemeral branch, we want to squash each pending PR into a commit on top of main.
+
+Assuming this repo is the `origin` remote, this looks something like:
```
Comment (njhill) [inline TGIS_README.md:66]: Should we add a step here to tag the existing `ephemeral` branch before overwriting?

**Review by njhill — COMMENTED** [review]
Thanks a lot for this @joerunde!

File: TGIS_README.md (hunk 3 above)
Comment (joerunde) [reply TGIS_README.md:62]: Yeah, `release` seems better. I'll rename

**Comment by joerunde** [comment]
snuck in some formatting fixes so we can have some ✅ s on `main` :D 

Thanks for the review @njhill!

**Comment by joerunde** [comment]
(jk, running the `format.sh` script doesn't magically solve al the format errors)

File: vllm/entrypoints/grpc/grpc_server.py [hunk 5]
```diff
@@ -2,31 +2,37 @@
...
+                                                     BatchedTokenizeResponse,
+                                                     DecodingMethod,
+                                                     GenerationResponse,
+                                                     ModelInfoRequest,
+                                                     ModelInfoResponse,
+                                                     Parameters,
+                                                     ResponseOptions,
+                                                     SingleGenerationRequest,
+                                                     StopReason, TokenInfo,
+                                                     TokenizeResponse)
```
Comment (njhill) [inline vllm/entrypoints/grpc/grpc_server.py:30]: yeah this makes me a bit sad

**Review by njhill — APPROVED** [review]
Thanks @joerunde!

File: vllm/entrypoints/grpc/grpc_server.py (hunk 5 above)
Comment (joerunde) [reply vllm/entrypoints/grpc/grpc_server.py:30]: yeah I'm not really sure what was going on- `format.sh` did this but then the `yapf` check failed on it :/

**Comment by joerunde** [comment]
just gonna go ahead and merge so we have it, we can always update formatting more later

**Pull request merged**